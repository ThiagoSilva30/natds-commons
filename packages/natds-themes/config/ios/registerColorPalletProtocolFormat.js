import fs from 'fs';
import path from 'path';
import Handlebars from 'handlebars';

const registerProtocolColorPalletFormat = () => {
  const html = fs.readFileSync(path.resolve(__dirname, './templates/colorPalette.hbs')).toString();
  const htmlTemplate = Handlebars.compile(html);

  const formatter = dictionary => htmlTemplate({
    color: dictionary.properties.color,
  });

  return {
    formatter,
    name: 'ios-swift/protocol-colors-palette.swift',
  };
};

export default registerProtocolColorPalletFormat;
