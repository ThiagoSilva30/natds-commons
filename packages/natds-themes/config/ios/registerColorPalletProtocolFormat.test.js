import path from 'path';
import registerColorPalletProtocolFormat from './registerColorPalletProtocolFormat';

const dictionary = {
  properties: {
    color: {
      primary: {
        attributes: {},
        name: 'colorPrimary',
        original: {
          value: '#F091C9',
        },
        path: [
          'color',
          'primary',
        ],
        value: '#F091C9',
      },
    },
  },
};

describe('registerColorPalletProtocolFormat', () => {
  it('should return the register config', () => {
    const register = registerColorPalletProtocolFormat();

    expect(typeof register.formatter).toBe('function');
    expect(register.name).toBe('ios-swift/protocol-colors-palette.swift');
  });

  it('should return the correct formatter template', () => {
    const register = registerColorPalletProtocolFormat();

    const output = register.formatter(dictionary);

    expect(output).toMatchFile(path.join(__dirname, '__file_snapshots__/registerColorPalletProtocolFormat.snap.swift'));
  });
});
