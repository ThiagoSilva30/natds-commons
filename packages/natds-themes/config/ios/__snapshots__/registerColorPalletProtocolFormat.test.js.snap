// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerColorPalletProtocolFormat should return the correct formatter template 1`] = `
"protocol ColorPalette {
  var primary: String { get }
}"
`;
